drivers/ng_at86rf2xx/ng_at86rf2xx.c

```diff
@@ -101,28 +102,39 @@ void ng_at86rf2xx_reset(ng_at86rf2xx_t *dev)
     /* set short and long address */
 #if CPUID_ID_LEN
     cpuid_get(cpuid);
-#if CPUID < 8
+#if CPUID_ID_LEN < 8
     /* in case CPUID_ID_LEN < 8, fill missing bytes with zeros */
     for (int i = CPUID_ID_LEN; i < 8; i++) {
         cpuid[i] = 0;
     }
+    /* copy and set long address */
+    memcpy(&addr_long, cpuid, 8);
 #else
-    for (int i = 8; i < CPUID_ID_LEN; i++) {
-        cpuid[i & 0x07] ^= cpuid[i];
-    }
+    /* for consistency taken from OpenWSN, https://github.com/openwsn-berkeley/openwsn-fw,
+     * bsp/boards/iot-lab_M3/eui64.c */
+    const uint8_t const *luid = (const uint8_t *const) &(cpuid[CPUID_ID_LEN - 8]);
+
+    /* Pseudo-OUI */
+    addr_long.uint8[0] = cpuid[3];
+    addr_long.uint8[1] = luid[7];
+    addr_long.uint8[2] = cpuid[0];
+    addr_long.uint8[3] = cpuid[1];
+    /* some variable bytes in the register */
+    addr_long.uint8[4] = luid[6];
+    addr_long.uint8[5] = cpuid[2];
+
+    addr_long.uint8[6] = (luid[4] | (luid[6] << 7));
+    addr_long.uint8[7] = luid[5];
 #endif
     /* make sure we mark the address as non-multicast and not globally unique */
     cpuid[0] &= ~(0x01);
     cpuid[0] |= 0x02;
-    /* copy and set long address */
-    memcpy(&addr_long, cpuid, 8);
-    ng_at86rf2xx_set_addr_long(dev, addr_long);
+    ng_at86rf2xx_set_addr_long(dev, addr_long.uint64);
     /* now compress the long addr to form the short address */
     for (int i = 2; i < 8; i++) {
         cpuid[i & 0x01] ^= cpuid[i];
     }
-    memcpy(&addr_short, cpuid, 2);
-    ng_at86rf2xx_set_addr_short(dev, addr_short);
+    ng_at86rf2xx_set_addr_short(dev, addr_long.uint16[3]);
```

Comment:
Okay, then leave it as it is. Should I PR a patch for xbee? To this branch or to master?